tests/osaka/eip7692_eof_v1/eip7873_tx_create/test_txcreate_validates.py

```diff
@@ -0,0 +1,229 @@
+"""Test bad TXCREATE cases."""
+
+from enum import Enum, auto, unique
+from typing import Tuple
+
+import pytest
+
+from ethereum_test_base_types import Bytes
+from ethereum_test_base_types.base_types import Address, Hash
+from ethereum_test_tools import (
+    Account,
+    Alloc,
+    Environment,
+    EVMCodeType,
+    StateTestFiller,
+    Transaction,
+    compute_eofcreate_address,
+)
+from ethereum_test_tools.vm.opcode import Opcodes as Op
+from ethereum_test_types.eof.v1 import Container, Section
+from ethereum_test_vm.bytecode import Bytecode
+from tests.prague.eip7702_set_code_tx.spec import Spec
+
+from .. import EOF_FORK_NAME
+from ..eip7620_eof_create.helpers import (
+    slot_a,
+    slot_b,
+    slot_code_worked,
+    smallest_initcode_subcontainer,
+    smallest_runtime_subcontainer,
+    value_code_worked,
+)
+from .spec import TXCREATE_FAILURE
+
+REFERENCE_SPEC_GIT_PATH = "EIPS/eip-7873.md"
+REFERENCE_SPEC_VERSION = "1115fe6110fcc0efc823fb7f8f5cd86c42173efe"
+
+pytestmark = pytest.mark.valid_from(EOF_FORK_NAME)
+
+
+@unique
+class ValidatedCode(Enum):
+    """Kinds of valid/invalid EOF."""
+
+    LEGACY = auto()
+    EF = auto()
+    EOFV1_RUNTIME = auto()
+    EOFV1_RUNTIME_INVALID = auto()
+    EOFV1_INITCODE = auto()
+    EOFV2 = auto()
+    DELEGATION = auto()
+    SUBCONTAINER_INVALID = auto()
+
+    def bytecode(self) -> Bytecode | Container | Bytes:
+        """Bytecode for the code to validate."""
+        match self:
+            case ValidatedCode.LEGACY:
+                return Op.STOP
+            case ValidatedCode.EF:
+                return Bytes("0xEF")
+            case ValidatedCode.EOFV1_RUNTIME:
+                return smallest_runtime_subcontainer
+            case ValidatedCode.EOFV1_RUNTIME_INVALID:
+                return Container.Code(Op.ADD)
+            case ValidatedCode.EOFV1_INITCODE:
+                return smallest_initcode_subcontainer
+            case ValidatedCode.EOFV2:
+                return Bytes("0xEF0002")
+            case ValidatedCode.DELEGATION:
+                return Bytes(Spec.DELEGATION_DESIGNATION + Bytes("".join(20 * ["ab"])))
+            case ValidatedCode.SUBCONTAINER_INVALID:
+                return Container(
+                    sections=[
+                        Section.Code(Op.RETURNCODE[0](0, 0)),
+                        Section.Container(Container.Code(Op.ADD)),
+                    ]
+                )
+
+    def bytecode_hash(self) -> Hash:
+        """Hash of the bytecode to validate."""
+        bytecode = self.bytecode()
+        if isinstance(bytecode, Bytecode):
+            return bytecode.keccak256()
+        elif isinstance(bytecode, Container):
+            return bytecode.hash
+        elif isinstance(bytecode, Bytes):
+            return bytecode.keccak256()
+        else:
+            raise ValueError("Unknown bytecode type")
```

Comment:
I tried hard to not have this, but I failed to return something from `bytecode` which hashes allright and is also fine with mypy. Any Python magic to make this go away?

Reply:
yup, I was clearly overthinking this, thx :)!